### PR TITLE
Add number of attempts students can try out in an assignment

### PR DESCRIPTION
**Pull request by TimeString**
In this pull request, I introduce a new feature, instructors can specify how many attempts a student / team can try in an assignment. Students cannot make any further submission if they reach the limit.

screen shot 2017-11-03 at 1 10 22 am

screen shot 2017-11-03 at 1 10 04 am

screen shot 2017-11-03 at 1 10 49 am

screen shot 2017-11-03 at 1 11 26 am

screen shot 2017-11-03 at 1 10 49 am

### CODE REVIEW

**Comment by PARanOiA1120** [comment]
Does this limit apply to all submission scopes? Like if the limit is 4, then the students can only submit 4 times no matter which scope they are testing with?

File: embed_grader/serapis/forms/assignment_forms.py [hunk 1]
```diff
@@ -86,23 +94,45 @@ def __init__(self, *args, **kwargs):
...
                 required=True,
                 widget=forms.RadioSelect,
                 choices=AssignmentForm.TEAM_CHOICES,
                 initial=initial_team_choice_val,
         )
-        self.fields['num_max_team_members'] = forms.IntegerField(
+        self.fields['max_num_team_members'] = forms.IntegerField(
                 required=False,
                 initial=initial_num_member_val,
                 validators=[MinValueValidator(2)]
```
Comment (PARanOiA1120) [inline embed_grader/serapis/forms/assignment_forms.py]: Do we want a comma here? I think we should make it consistent, either to add the comma or remove the comma from line 107

File: embed_grader/serapis/forms/assignment_forms.py [hunk 2]
```diff
@@ -86,23 +94,45 @@ def __init__(self, *args, **kwargs):
...
+        self.fields['submission_limit_choice'] = forms.ChoiceField(
+                required=True,
+                widget=forms.RadioSelect,
+                choices=AssignmentForm.SUBMISSION_LIMIT_CHOICES,
+                initial=initial_submission_limit_choice_val,
+        )
+        self.fields['max_num_submissions'] = forms.IntegerField(
+                required=False,
+                initial=initial_num_submission_val,
+                validators=[MinValueValidator(1)]
```
Comment (PARanOiA1120) [inline embed_grader/serapis/forms/assignment_forms.py]: Same as last comment.

**Comment by TimeString** [comment]
Regarding the submission scope, that is a good point! I did think about this but forgot to change it. If there is a limit, the scope will only be debug+public+feedback (or debug+public+feedback+hidden for instructor mode. Wiil change it now.

File: embed_grader/serapis/forms/assignment_forms.py (hunk 1 above)
Comment (TimeString) [reply embed_grader/serapis/forms/assignment_forms.py]: Good catch! Always put a comment when parameters are in several lines

**Pull request merged**